### PR TITLE
feat: implement toggle_fullscreen and hide for Wayland

### PR DESCRIPTION
**Pull request by aliaksandr-trush**

### CODE REVIEW

File: window/src/os/wayland/window.rs [hunk 1]
```diff
@@ -1253,6 +1267,13 @@ impl WaylandState {
                     state |= WindowState::MAXIMIZED;
                 }
 
+                // For MEXIMIZED and FULL_SCREEN window configure contains Windowed size.
```
Comment (mbwilding) [inline window/src/os/wayland/window.rs:1270]: MAXIMIZED*

**Review by mbwilding — APPROVED** [review]

**Comment by emilnymann** [comment]
I switched to `wezterm-git` on Arch today to escape an unrelated visual glitch, but now this issue is pretty much preventing me from using Wezterm at all. What's the bottleneck on this being merged, apart from maintainer approval? Is there anything anyone can do to move it along, other than maintainer or PR author?

**Comment by mbwilding** [comment]
Just built this PR, works as expected. Still getting weird double up characters when enabling wayland mode in the config. But the PR does what it says it does. Can we please get some eyes on this @wez. Thanks.

**Comment by wez** [comment]
Thanks!

**Pull request closed without merging**